src/main/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -30,57 +30,63 @@
     private int reviews = 0;
     private long totalProd = 0;
     private long totalUsers = 0;
-    HashMap<String, Long> productsTotal = new HashMap<String, Long>();
-    HashMap<Long, String> productsTotalReverse = new HashMap<Long, String>();
-    HashMap<String, Long> usersTotal = new HashMap<String, Long>();
-    ArrayList<String> products = new ArrayList<String>();
-    ArrayList<String> users = new ArrayList<String>();
-    ArrayList<String> scores = new ArrayList<String>();
+    HashMap<String, Long> productsTotal = new HashMap<>();
+    HashMap<Long, String> productsTotalReverse = new HashMap<>();
+    HashMap<String, Long> usersTotal = new HashMap<>();
+    List<String> products = new ArrayList<>();
+    List<String> users = new ArrayList<>();
+    List<String> scores = new ArrayList<>();
     PrintWriter out = new PrintWriter(new FileWriter("movies.csv"));
 
     public MovieRecommender (String filename) throws IOException{
         this.filename = filename;
         getData();
     }
 
-    public void getData() throws IOException{
-        String line;
-        String idProduct="";
-        String idUser="";
-        String score;
-        InputStream fileStream = new FileInputStream(filename);
-        InputStream gzipStream = new GZIPInputStream(fileStream);
-        Reader decoder = new InputStreamReader(gzipStream, "UTF-8");
-        BufferedReader buffered = new BufferedReader(decoder);
+    public void getData(){
+        try{
+            String line;
+            String idProduct="";
+            String idUser="";
+            String score;
+            InputStream fileStream = new FileInputStream(filename);
+            InputStream gzipStream = new GZIPInputStream(fileStream);
+            Reader decoder = new InputStreamReader(gzipStream, "UTF-8");
+            BufferedReader buffered = new BufferedReader(decoder);
 
-        while ((line = buffered.readLine()) != null){
-            if(line.indexOf("product/productId:") != -1){
-                reviews++;
-                idProduct = line.replace("product/productId: ","");
-                if(!productsTotal.containsKey(idProduct)){
-                    productsTotal.put(idProduct, totalProd);
-                    totalProd++;
+            while ((line = buffered.readLine()) != null){
+                if(line.indexOf("product/productId:") != -1){
+                    reviews++;
+                    idProduct = line.replace("product/productId: ","");
+                    if(!productsTotal.containsKey(idProduct)){
+                        productsTotal.put(idProduct, totalProd);
+                        totalProd++;
```

Comment:
I see that you have a counter here to get the total of products. What do you think about using the size of your map to count that?